elasticsearch-core/src/main/scala/com/sumologic/elasticsearch/restlastic/dsl/MappingDsl.scala

```diff
@@ -39,11 +39,15 @@ trait MappingDsl extends DslCommons {
   sealed trait IndexType {
     val rep: String
   }
-
+  
   case object NotAnalyzedIndex extends IndexType {
     val rep = "not_analyzed"
   }
 
+  case object NotIndexedIndex extends IndexType {
```

Comment:
We would like to maintain certain fields unindexed such that it is available when we retrieve the document, but do not take space and cpu when indexing.